packages/graphql/tests/integration/issues/350.int.test.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Driver } from "neo4j-driver";
+import { graphql } from "graphql";
+import { gql } from "apollo-server";
+import { generate } from "randomstring";
+import neo4j from "../neo4j";
+import { Neo4jGraphQL } from "../../../src/classes";
+
+describe("https://github.com/neo4j/graphql/issues/350", () => {
+    let driver: Driver;
+
+    beforeAll(async () => {
+        driver = await neo4j();
+        const session = driver.session();
+        try {
+            await session.run(`
```

Comment:
We haven't done this elsewhere... But honestly, I'm not against it, our integration tests leave the database a real mess.

Reply:
I'm not against it either, something for another time but I wonder if we can add this as a global post hook that runs once after all the tests are complete.

Reply:
I originally added this in a `beforeEach` but moved it to `beforeAll` since I only had one test. I wanted this to be able to use the `toHaveLength` method to ensure I was getting back the correct number of records. In order to ensure this I had to clear the database.

It would be nice to have a utility function that clears the database if needed. Something like:
```typescript
const clearDatabase = async (driver: Driver) => {
  const session = driver.session()
  try {
    await session.run(`
      MATCH (n)
      DETACH DELETE n
    `)
  } finally {
    await session.close()
  }
}
```

You could then call this in a `beforeEach` hook when writing tests if needed and in a global `afterAll` hook to clean up the database when done.

Reply:
So, I just checked out this branch and I just cannot get the tests (in other suites) to pass. When I remove this clear-down, all of the tests pass except for this one. 

I know it's not right and our integration tests shouldn't be so co-dependent, but for the sake of getting this merged, can you change this test so that it can work without clearing the database down? 

For example, you could give the types unique names such as "Post350" and "Comment350", and then do `MATCH (p:Post350) DETACH DELETE (p)` (and the same for comments) before the tests, instead of clearing the whole database.

Reply:
Sorry; fixed in https://github.com/neo4j/graphql/pull/359/commits/a8ca8f7b4cfb7404ecde4cd0838560f4056013c5. I was having trouble with `toHaveLength`. I ultimately abandoned it, but it would be nice to have cleaner tests.